### PR TITLE
Cleanup ReviewBeforeRelease

### PR DESCRIPTION
**Pull request by zoewangg**

### CODE REVIEW

File: utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java [hunk 1]
```diff
@@ -107,7 +106,7 @@
  * <p>#ThreadSafe#</p>
  * @see java.lang.String
  */
-@ReviewBeforeRelease("Remove the methods we don't end up using (and we've removed software.amazon.awssdk.core.util.StringUtils).")
+//TODO Remove the methods we don't end up using (and we've removed software.amazon.awssdk.core.util.StringUtils).
```
Comment (varunnvs92) [inline utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java]: we can remove it? I don't see any unused methods

File: utils/src/main/java/software/amazon/awssdk/utils/Validate.java [hunk 2]
```diff
@@ -47,7 +46,7 @@
  * <p>#ThreadSafe#</p>
  * @see java.lang.String#format(String, Object...)
  */
-@ReviewBeforeRelease("Remove the methods we don't end up using (and software.amazon.awssdk.core.util.ValidationUtils).")
+//TODO: Remove the methods we don't end up using (and software.amazon.awssdk.core.util.ValidationUtils).
```
Comment (varunnvs92) [inline utils/src/main/java/software/amazon/awssdk/utils/Validate.java]: There is one unused method. Can we remove it now.

File: utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java (hunk 1 above)
Comment (zoewangg) [reply utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java]: Yeah, will do

File: utils/src/main/java/software/amazon/awssdk/utils/Validate.java (hunk 2 above)
Comment (zoewangg) [reply utils/src/main/java/software/amazon/awssdk/utils/Validate.java]: Yeah, will do

**Pull request merged**